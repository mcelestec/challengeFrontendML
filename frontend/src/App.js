import React from 'react'
import { BrowserRouter, Routes, Route, Link } from 'react-router-dom';
import SearchBar from './components/search/SearchBar';
import HomePage from './pages/HomePage';
import ErrorPage from './pages/ErrorPage';
import ProductDetailPage from './pages/products/ProductDetailPage';
import SearchResultPage from './pages/search/SearchResultPage';


function App() {

  return (
    <BrowserRouter>
      <SearchBar />

      <Routes>
        <Route exact path='/' element={<HomePage />} />
        <Route path='/items/:id' element={<ProductDetailPage />} />
        <Route path='/items' element={<SearchResultPage />} />
        <Route path='*' element={
          <ErrorPage >
            <h4>Parece que esta página no existe</h4>
            <Link to="/" aria-hidden>Ir a la página principal</Link>
          </ErrorPage>
        } />
      </Routes>
    </BrowserRouter>
  );
}

export default App;
