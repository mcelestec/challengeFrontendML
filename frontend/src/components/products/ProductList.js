import React from 'react'
import ProductItem from './ProductItem'
import PropTypes from 'prop-types';

const ProductList = ({ items }) => {
    return (
        <section className="product-list__container">
            <ul>
                {
                    items.map(item =>
                        <li key={item.id}>
                            <ProductItem item={item} />
                        </li>
                    )
                }
            </ul>
        </section>
    )
}

ProductList.propTypes = {
    items: PropTypes.arrayOf(
        ProductItem.propTypes.item
    ).isRequired
}

export default ProductList