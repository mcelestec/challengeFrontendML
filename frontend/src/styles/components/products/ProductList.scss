.product-list__container {
    background-color: $white;
    border-radius: $border-radius;
    display: flex;
    flex-direction: column;
    padding: 0 16px;

    @include lg {
        margin: 0;
    }

    ul {
        padding: 0;

        li {
            list-style: none;
            border-bottom: 1px solid $ligth-gray;

            &:last-child {
                border-bottom: none;
            }
        }
    }

}