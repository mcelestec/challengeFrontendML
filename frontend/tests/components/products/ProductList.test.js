import { render } from "@testing-library/react";
import App from "../../../src/App";
import ProductList from "../../../src/components/products/ProductList";

describe('Test: probando <ProductList />', () => {
    const items = [{
        "id": "MLA1141217077",
        "title": "Casa 3 Ambientes Mas Estudio Balvanera Once",
        "address": "Capital Federal",
        "price": {
            "currency": "ARS",
            "amount": 59000,
            "decimals": 0
        },
        "picture": "http://http2.mlstatic.com/D_946381-MLA50220484139_062022-I.jpg",
        "free_shipping": false,
        "condition": "new",
        "sold_quantity": 5
    },
    {
        "id": "MLA3341217077",
        "title": "Casa 2 ambientes",
        "address": "Santa Fe",
        "price": {
            "currency": "ARS",
            "amount": 79000,
            "decimals": 0
        },
        "picture": "http://http2.mlstatic.com/D_946381-MLA50220484139_062022-I.jpg",
        "free_shipping": false,
        "condition": "new",
        "sold_quantity": 0
    }];

    test('Llamada a ProductList: debe hacer match con el snapshot', () => {
        const { container } = render(<ProductList items={items} />, { wrapper: App });
        expect(container).toMatchSnapshot();
    }
    );

});