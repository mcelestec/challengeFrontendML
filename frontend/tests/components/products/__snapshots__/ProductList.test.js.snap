// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test: probando <ProductList /> Llamada a ProductList: debe hacer match con el snapshot 1`] = `
<div>
  <header
    class="search-bar__container"
  >
    <nav>
      <a
        aria-hidden="true"
        href="/"
      >
        Mercado Libre Logo
      </a>
      <div
        class="search-box__container"
      >
        <form>
          <input
            name="filterText"
            placeholder="Nunca dejes de buscar"
            type="text"
            value=""
          />
          <input
            type="submit"
          />
        </form>
      </div>
    </nav>
  </header>
</div>
`;
